src/snowflake/cli/plugins/git/commands.py

```diff
@@ -0,0 +1,239 @@
+
+app = SnowTyper(
+    name="git",
+    help="Manages git repositories in Snowflake.",
+)
+log = logging.getLogger(__name__)
+
+
+def _repo_path_argument_callback(path):
+    if not is_stage_path(path):
+        raise ClickException(
+            "REPOSITORY_PATH should be a path to git repository stage with scope provided."
+            " For example: @my_repo/branches/main/"
+        )
+    return path
+
+
+RepoNameArgument = identifier_argument(sf_object="git repository", example="my_repo")
+RepoPathArgument = typer.Argument(
+    metavar="REPOSITORY_PATH",
+    help=(
+        "Path to git repository stage with scope provided."
+        " Path to the repository root must end with '/'."
+        " For example: @my_repo/branches/main/"
+    ),
+    callback=_repo_path_argument_callback,
+)
+PatternOption = pattern_option(
+    help_example='`list-files --pattern ".*\.txt"` lists all files with .txt extension',
+)
+
+
+def _object_exists(object_type, identifier):
+    try:
+        ObjectManager().describe(
+            object_type=object_type.value.cli_name,
+            name=identifier,
+        )
+        return True
+    except ProgrammingError:
+        return False
+
+
+def _assure_repository_does_not_exist(repository_name: str) -> None:
+    if _object_exists(ObjectType.GIT_REPOSITORY, repository_name):
+        raise ClickException(f"Repository '{repository_name}' already exists")
+
+
+def _validate_origin_url(url: str) -> None:
+    if not url.startswith("https://"):
+        raise ClickException("Url address should start with 'https'")
+
+
+@app.command("setup", requires_connection=True)
+def setup(
+    repository_name: str = RepoNameArgument,
+    **options,
+) -> CommandResult:
+    """
+    Sets up a git repository object.
+
+    You will be prompted for:
+
+    * url - address of repository to be used for git clone operation
+
+    * secret - Snowflake secret containing authentication credentials. Not needed if origin repository does not require
+    authentication for RO operations (clone, fetch)
+
+    * API integration - object allowing Snowflake to interact with git repository.
+    """
+    _assure_repository_does_not_exist(repository_name)
+    manager = GitManager()
+
+    url = typer.prompt("Origin url")
+    _validate_origin_url(url)
+
+    secret = {}
+    secret_needed = typer.confirm("Use secret for authentication?")
+    if secret_needed:
+        secret_name = f"{repository_name}_secret"
+        secret_name = typer.prompt(
+            "Secret identifier (will be created if not exists)", default=secret_name
+        )
+        secret = {"name": secret_name}
+        if _object_exists(ObjectType.SECRET, secret_name):
+            cli_console.step(f"Using existing secret '{secret_name}'")
+        else:
+            cli_console.step(f"Secret '{secret_name}' will be created")
+            secret["username"] = typer.prompt("username")
+            secret["password"] = typer.prompt("password/token", hide_input=True)
```

Comment:
Any particular reason to keep this in dict instead of variables like `username` and `password`?

Reply:
Explicit over implicit 😉

Reply:
done